mindmaps-graql/src/main/java/io/mindmaps/graql/internal/reasoner/predicate/AtomBase.java

```diff
@@ -74,6 +74,9 @@ public AtomBase(AtomBase a) {
         typeId = varData.getValue();
     }
 
+    @Override
+    public Atomic clone(){ return null;}
```

Comment:
A method that returns `null` always ? Why are you implementing it here ? Maybe `AtomBase` could be abstract and that way you would not need this method. Unless you really need `clone()` to return `null` for something.

If you never initialise AtomBase then it should be abstract.